enable default appsettings for self hosted installs

kspearrin opened the pull request:
Our self-hosted installs have a lot of app setting configurations that are boilerplate and don't really need to be configured by the user. This PR reduces the need for these settings to be set by the user (as env vars) and defaults them to sensible defaults. Users still have the ability to explicitly set them, if desired, and existing installs where they are already set will still continue to work in a backward-compatible way.

joseph-flinn commented:
Looks good from a DevOps perspective. I think we might want to add all of the env vars that have been removed to the documentation: https://bitwarden.com/help/article/environment-variables/

MGibson1 commented:
Spoke with Kyle offline, but the failing tests are circular dependencies.

Since the circular parameters are being set in the `GlobalSettings` constructor, I believe they'll be fixed by a Customizer for `GlobalSettings` that does not set `BaseServiceUri`, `Attachment`, `Send`, or `DataProtection`. This would be similar to how `UserId` is not set in the [`OrganizationCipher` customization](https://github.com/bitwarden/server/blob/c1ceeace956444ffb18d31e1b7bd661a4560cb0f/test/Core.Test/AutoFixture/CipherFixtures.cs#L9-L21).

I would also recommend a test just ensuring the default `GlobalSettings` matches the current expected boilerplate ones

kspearrin commented:
updated setters to use arrow function and fixed tests

cscharf commented:
> updated setters to use arrow function and fixed tests

Looks good, just one more build issue it appears:

```
Error: D:\a\server\server\test\Core.Test\Services\OrganizationServiceTests.cs(46,49): error CS0104: 'GlobalSettings' is an ambiguous reference between 'Bit.Core.Test.AutoFixture.GlobalSettings' and 'Bit.Core.Settings.GlobalSettings' [D:\a\server\server\test\Core.Test\Core.Test.csproj]
Error: D:\a\server\server\test\Core.Test\Services\OrganizationServiceTests.cs(108,49): error CS0104: 'GlobalSettings' is an ambiguous reference between 'Bit.Core.Test.AutoFixture.GlobalSettings' and 'Bit.Core.Settings.GlobalSettings' [D:\a\server\server\test\Core.Test\Core.Test.csproj]
```

kspearrin commented:
> 
> 
> > updated setters to use arrow function and fixed tests
> 
> Looks good, just one more build issue it appears:
> 
> ```
> Error: D:\a\server\server\test\Core.Test\Services\OrganizationServiceTests.cs(46,49): error CS0104: 'GlobalSettings' is an ambiguous reference between 'Bit.Core.Test.AutoFixture.GlobalSettings' and 'Bit.Core.Settings.GlobalSettings' [D:\a\server\server\test\Core.Test\Core.Test.csproj]
> Error: D:\a\server\server\test\Core.Test\Services\OrganizationServiceTests.cs(108,49): error CS0104: 'GlobalSettings' is an ambiguous reference between 'Bit.Core.Test.AutoFixture.GlobalSettings' and 'Bit.Core.Settings.GlobalSettings' [D:\a\server\server\test\Core.Test\Core.Test.csproj]
> ```

Fixed
